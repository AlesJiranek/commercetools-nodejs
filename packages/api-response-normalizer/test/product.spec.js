import normalizeProduct from '../src/product'
import productAResponse from './fixtures/product-1.json'

describe('when the response is from `/product/:id`', () => {
  it('should normalize response', () => {
    expect(normalizeProduct(productAResponse)).toMatchSnapshot()
  })
})
