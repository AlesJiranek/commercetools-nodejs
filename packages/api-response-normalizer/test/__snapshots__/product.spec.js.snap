// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the response is from \`/product/:id\` should normalize tiny response 1`] = `
Object {
  "entities": Object {
    "categories": Object {
      "category-id-1": Object {
        "id": "category-id-1",
      },
      "category-id-2": Object {
        "id": "category-id-2",
      },
    },
    "channels": Object {
      "channel-id-1": Object {
        "custom": Object {
          "type": Object {
            "obj": "custom-id-1",
          },
        },
        "id": "channel-id-1",
      },
      "channel-id-2": Object {
        "custom": Object {
          "type": Object {
            "obj": "custom-id-2",
          },
        },
        "id": "channel-id-2",
      },
    },
    "customerGroups": Object {
      "customer-group-id-1": Object {
        "id": "customer-group-id-1",
      },
      "customer-group-id-2": Object {
        "id": "customer-group-id-2",
      },
    },
    "productDiscounts": Object {
      "product-discount-id-1": Object {
        "id": "product-discount-id-1",
      },
    },
    "productTypes": Object {
      "product-type-id-1": Object {
        "id": "product-type-id-1",
      },
    },
    "products": Object {
      "product-id-1": Object {
        "id": "product-id-1",
        "masterData": Object {
          "current": Object {
            "categories": Array [
              Object {
                "obj": "category-id-1",
              },
              Object {
                "obj": "category-id-2",
              },
            ],
            "masterVariant": Object {
              "assets": Array [
                Object {
                  "custom": Object {
                    "type": Object {
                      "obj": "custom-id-2",
                    },
                  },
                },
              ],
              "prices": Array [
                Object {
                  "channel": Object {
                    "obj": "channel-id-2",
                  },
                  "custom": Object {
                    "type": Object {
                      "obj": "custom-id-2",
                    },
                  },
                  "customerGroup": Object {
                    "obj": "customer-group-id-1",
                  },
                  "discounted": Object {
                    "discount": Object {
                      "obj": "product-discount-id-1",
                    },
                  },
                },
              ],
              "scopedPrice": Object {
                "channel": Object {
                  "obj": "channel-id-1",
                },
                "custom": Object {
                  "type": Object {
                    "obj": "custom-id-3",
                  },
                },
                "customerGroup": Object {
                  "obj": "customer-group-id-2",
                },
                "discounted": Object {
                  "discount": Object {
                    "obj": "product-discount-id-1",
                  },
                },
              },
            },
            "variants": Array [
              Object {
                "assets": Array [
                  Object {
                    "custom": Object {
                      "type": Object {
                        "obj": "custom-id-2",
                      },
                    },
                  },
                ],
                "prices": Array [
                  Object {
                    "channel": Object {
                      "obj": "channel-id-1",
                    },
                    "custom": Object {
                      "type": Object {
                        "obj": "custom-id-2",
                      },
                    },
                    "customerGroup": Object {
                      "obj": "customer-group-id-1",
                    },
                    "discounted": Object {
                      "discount": Object {
                        "obj": "product-discount-id-1",
                      },
                    },
                  },
                ],
                "scopedPrice": Object {
                  "channel": Object {
                    "obj": "channel-id-2",
                  },
                  "custom": Object {
                    "type": Object {
                      "obj": "custom-id-3",
                    },
                  },
                  "customerGroup": Object {
                    "obj": "customer-group-id-2",
                  },
                  "discounted": Object {
                    "discount": Object {
                      "obj": "product-discount-id-1",
                    },
                  },
                },
              },
            ],
          },
        },
        "productType": Object {
          "obj": "product-type-id-1",
        },
      },
    },
    "types": Object {
      "custom-id-1": Object {
        "id": "custom-id-1",
      },
      "custom-id-2": Object {
        "id": "custom-id-2",
      },
      "custom-id-3": Object {
        "id": "custom-id-3",
      },
    },
  },
  "result": "product-id-1",
}
`;
